bitsandbytes/functional.py

```diff
@@ -421,41 +414,36 @@ def create_quantile_map(A, total_bits=8):
     return q
 
 
+# TODO: Deprecate
 def get_special_format_str():
-    if not torch.cuda.is_available():
-        return "col_turing"
-    major, _minor = torch.cuda.get_device_capability()
-    if major <= 7:
-        return "col_turing"
-    if major == 8:
-        return "col_ampere"
-    return "col_turing"
+    return "row"
 
 
-def is_on_gpu(tensors):
+def is_on_gpu(tensors: Iterable[torch.Tensor]):
     on_gpu = True
     gpu_ids = set()
+
     for t in tensors:
-        if t is None:
-            continue  # NULL pointers are fine
-        is_paged = getattr(t, "is_paged", False)
-        on_gpu &= t.device.type == "cuda" or is_paged
-        if not is_paged:
+        # NULL pointers and paged tensors are OK.
+        if t is not None and not getattr(t, "is_paged", False):
+            on_gpu &= t.is_cuda
             gpu_ids.add(t.device.index)
+
     if not on_gpu:
-        raise TypeError(
+        raise RuntimeError(
             f"All input tensors need to be on the same GPU, but found some tensors to not be on a GPU:\n {[(t.shape, t.device) for t in tensors]}",
         )
+
     if len(gpu_ids) > 1:
-        raise TypeError(
+        raise RuntimeError(
             f"Input tensors need to be on the same GPU, but found the following tensor and device combinations:\n {[(t.shape, t.device) for t in tensors]}",
         )
     return on_gpu
 
 
-def get_tensor_stream(tensor: Tensor) -> torch.cuda.Stream:
-    stream = torch.cuda.current_stream(tensor.device)
-    return stream
+def get_tensor_stream(tensor: Tensor) -> ct.c_void_p:
+    # We use the raw stream for performance reasons.
+    return ct.c_void_p(torch._C._cuda_getCurrentRawStream(tensor.device.index))
```

Comment:
There was far more overhead with constructing a `torch.cuda.Stream()` for this than I would have thought. Certainly more than I would have liked. So while not pretty, this is another performance improvement that reaches beyond int8.

Reply:
This changing the rtype for a public function, which is probably not OK.

I'd add a new `_get_raw_tensor_stream` function and use that...

Reply:
Thanks, fair point! I don't think we should have exposed this as a public facing API, so I've marked it deprecated and introduced a private `_get_tensor_stream` as the replacement.